cores/esp8266/uart.c

```diff
@@ -739,3 +740,39 @@ uart_get_debug()
 {
     return s_uart_debug_nr;
 }
+
+/*
+To start detection of baud rate with the UART the UART_AUTOBAUD_EN bit needs to be cleared and set. The ROM function uart_baudrate_detect() does this only once, so on a next call the UartDev.rcv_state is not equal to BAUD_RATE_DET. Instead of poking around in the UartDev struct with unknown effect, the UART_AUTOBAUD_EN bit is directly triggered by the function uart_detect_baudrate().
+*/
+int
+uart_detect_baudrate(int uart_nr)
+{
+    static bool doTrigger = true;
```

Comment:
This doTrigger var is inited to true, then set to false on first run. In order to reset it to true, the detection must actually succeed in a subsequent call.
There is no way to abort the detection process, or reset a detection considered to have failed.

Reply:
Off the top of my head, I see the following possibilities:
1. add an argument which resets doTrigger when true. The arg should default to false.
2. make doTrigger a global static, and implement a resetter function.

Reply:
This was done on purpose, to have only one function to do the detection. If no detection takes place between two subsequent calls to uart_detect_baudrate(), triggering is not necessary: the uart is still in detecting state, and the function returns 0. If (enough) data was presented to the uart, the correct baudrate is returned.
If you prefer a "reset", I suggest to make a void uart_start_detect_baudrate(int uart_nr), that does the triggering, and a int uart_detect_baudrate(uart_nr) that reads the baudrate.
What do you prefer?

Reply:
I don't know if I get you right... The uart starts detecting by clearing and setting the UART_AUTOBAUD_EN bit. I do not know if it stops detecting by only clearing the bit. But why do you want to stop the uart detecting? It is not consuming any processor time. Retrying is just: call uart_detect_baudrate() again. Giving up is just stop calling that function. The detection of the baudrate *does not change it*. In the case of multiple devices, if you want to detect some device D after another device B was detected uart_detect_baudrate() will return the rate of device B and reinit. If you don't need the device B baudrate, just call uart_detect_baudrate once, ignore the return value, and start detecting in a loop until device D comes up. Am I not getting you???

I will implement a hard restart, in uart.h and uart.c, just in case.

About your last suggestion, that would block the processor in waiting for the uart. What about two wrappers? A non blocking detectBaudrate() and a blocking detectBaudrate(time_t timeoutMillis)?